scripts/regions-population

```diff
@@ -8,29 +8,45 @@ DATA_PATH=${DATA_PATH:-data}
 # Creates a raster file data/populations/REGIONID.tif with the population values
 # of each REGIONID in the regions table
 
-echo "Precalculating population count per region"
-REGION_IDS="$(psql -d $POSTGRES_DB -U $POSTGRES_USER -h $POSTGRES_HOST -t -A -c 'SELECT id FROM regions WHERE total_population IS NULL;')"
+echo "Precalculating population aggregates per region"
+REGION_IDS="$(psql -d $POSTGRES_DB -U $POSTGRES_USER -h $POSTGRES_HOST -t -A -c 'SELECT id FROM regions WHERE total_population IS NULL OR max_population IS NULL;')"
 echo " Region ids: ${REGION_IDS}"
 POPULATION_FILE="${DATA_PATH}/KEN_popmap15_v2b.tif"
 
 mkdir -p ${DATA_PATH}/populations
 
 for id in $REGION_IDS; do
   TARGET=${DATA_PATH}/populations/${id}.tif
-  if [[ ! -e $TARGET ]]; then
+
+  if [[ ! -e $TARGET || -z `gdalinfo $TARGET | grep -e POPULATION_SUM -e POPULATION_MAX` ]]; then
+    TMP_FILE=`mktemp`
+
     echo " Warping population for ${id}"
     gdalwarp -q \
       -co "TILED=YES" -co "BLOCKXSIZE=128" -co "BLOCKYSIZE=128" \
       -crop_to_cutline \
       -cutline PG:"dbname=${POSTGRES_DB} host=${POSTGRES_HOST} user=${POSTGRES_USER} password=${POSTGRES_PASSWORD}" \
       -csql "SELECT the_geom FROM regions WHERE id = ${id};" \
-      $POPULATION_FILE $TARGET
+      $POPULATION_FILE $TMP_FILE
+
+    echo " Aggregating population for ${id}"
+    IFS=' ' read -a AGGREGATES <<< $(${BIN_PATH}/aggregate-population ${TMP_FILE})
+    POPULATION_SUM=${AGGREGATES[0]}
+    POPULATION_MAX=${AGGREGATES[1]}
+
+    echo " Normalizing raster file scale for ${id}"
+    gdal_translate -ot Byte -scale 0 $POPULATION_MAX 0 255 $TMP_FILE $TARGET
```

Comment:
Actually, if we save both the original and the translated raster, we would not need a VRT at all. But this can all wait until we rework the demand calculation.